fearless/Modules/Staking/StakingMain/StakingMainPresenter.swift

```diff
@@ -40,29 +40,37 @@ final class StakingMainPresenter {
             return
         }
 
-        var payoutPercentage: Decimal?
         do {
+
+            let monthlyReturn: Decimal
+            let yearlyReturn: Decimal
+
             if let calculator = calculator {
-                payoutPercentage = try calculator.calculateNetworkReturn(isCompound: false,
-                                                                             period: .year)
+                monthlyReturn = try calculator.calculateNetworkReturn(isCompound: true,
+                                                                      period: .month)
+                yearlyReturn = try calculator.calculateNetworkReturn(isCompound: true,
+                                                                     period: .year)
+            } else {
+                monthlyReturn = 0.0
+                yearlyReturn = 0.0
             }
-        } catch {
-            logger?.error("Can't calculate reward")
-        }
 
-        let curAmount = amount ?? 0.0
-        let reward = curAmount * (payoutPercentage ?? 0.0)
+            let monthlyViewModel = viewModelFactory
+                .createRewardViewModel(reward: (amount ?? 0.0) * monthlyReturn,
+                                       targetReturn: monthlyReturn,
+                                       priceData: priceData)
 
-        let monthlyViewModel = viewModelFactory.createMonthlyRewardViewModel(amount: curAmount,
-                                                                             reward: reward,
-                                                                             priceData: priceData)
+            let yearlyViewModel = viewModelFactory
+                .createRewardViewModel(reward: (amount ?? 0.0) * yearlyReturn,
+                                       targetReturn: yearlyReturn,
+                                       priceData: priceData)
```

Comment:
Looks like calculations are still incorrect: 
0.0002 KSM * 12 = 0.0024 KSM (expected 0.0036 KSM)
1,1% * 12 = 13,2% (expected 14,31%)
$0.06 * 12 = $0.72 (expected $0.83)
<img width="553" alt="image" src="https://user-images.githubusercontent.com/3176149/110383019-c8e1da00-806c-11eb-9687-203ae50384c5.png">

Reply:
Did you check compound case?

Reply:
Oh, I see now. In this case everything seems right. However, it might be confusing for a user. Let's discuss later